lib/Sema/CSApply.cpp

```diff
@@ -6574,6 +6574,20 @@ Expr *ExprRewriter::coerceToType(Expr *expr, Type toType,
                                  Optional<Pattern*> typeFromPattern) {
   auto &ctx = cs.getASTContext();
 
+  // Diagnose conversions to invalid function types that couldn't be performed
+  // beforehand because of placeholders.
+  if (auto *fnTy = toType->getAs<FunctionType>()) {
+    auto contextTy = cs.getContextualType(expr);
+    if (cs.getConstraintLocator(locator)->isForContextualType() &&
+        contextTy && contextTy->hasPlaceholder()) {
+      bool hadError = TypeChecker::diagnoseInvalidFunctionType(fnTy,
```

Comment:
Nit: Is this a valid formatting?

Reply:
I'll run it thru `clang-format` again just to make sure.

Reply:
Thank you!